src/libcore/option.rs

```diff
@@ -414,6 +479,19 @@ impl<T> Option<T> {
     /// assert_eq!(Some(4).unwrap_or_else(|| 2 * k), 4);
     /// assert_eq!(None.unwrap_or_else(|| 2 * k), 20);
     /// ```
+    #[rustc_const_unstable(feature = "const_option_match")]
+    #[cfg(not(bootstrap))]
+    #[inline]
+    #[stable(feature = "rust1", since = "1.0.0")]
+    pub const fn unwrap_or_else<F: FnOnce() -> T>(self, f: F) -> T {
```

Comment:
You show this function works in constants, but this should not be possible. It's not for `min_const_fn`, just for the `const_fn` feature gate, but still: We seem to have lost a qualification check somewhere along the line. cc @ecstatic-morse the function call to `f` should not work without `-Zunleash-the-miri-inside-of-you`

For this PR, please undo all constifications for functions that take generic things with trait bounds (other than `Sized`), these require https://github.com/rust-lang/rfcs/pull/2632 first

Reply:
Outside of `core`, the equivalent code [is rejected](https://play.rust-lang.org/?version=nightly&mode=debug&edition=2018&gist=adc74e171695de7c7436bcde1adeb043). Any idea what's going on here?

Reply:
Reverted constification for these functions in 99a885e -- should I leave this conversation unresolved why we work out why they weren't rejected, though?

Reply:
huh!? Even with the `const_fn` feature gate it still errors... that is worrysome. Yea leave it unresolved. I'll try to create a non-core repro